packages/dids/src/did.ts

```diff
@@ -267,29 +286,12 @@ export class DID {
     payload: T,
     options: CreateJWSOptions = {}
   ): Promise<DagJWS> {
-    if (this._client == null) throw new Error('No provider available')
-    if (this._id == null) throw new Error('DID is not authenticated')
-    if (this._capability) {
-      const exp = this._capability.p.exp
-      if (exp && Date.parse(exp) < Date.now()) {
-        throw new Error('Capability is expired, cannot create a valid signature')
-      }
-      const cacaoBlock = await CacaoBlock.fromCacao(this._capability)
-      const capCID = CID.asCID(cacaoBlock.cid)
-      if (!capCID) {
-        throw new Error(
-          `Capability CID of the JWS cannot be set to the capability payload cid as they are incompatible`
-        )
-      }
-      options.protected = options.protected || {}
-      options.protected.cap = `ipfs://${capCID?.toString()}`
-    }
-    const { jws } = await this._client.request('did_createJWS', {
-      did: this._id,
-      ...options,
-      payload,
+    return await createJWSUsing({
```

Comment:
You do not have to do `return await` here. And, `createJWS` could be unmarked `async`. Wrapping/unwrapping promises adds an overhead. Same for other `return await` clauses.